### PR TITLE
Cleanup of bone animation code

### PR DESCRIPTION
**Pull request by TmLev**
This pull request aims to:
* Add appropriate attributes for some methods.
* Use `nullptr` where possible instead of `0` / `NULL`.
* Adapt range-based for.
* Add `override` and `const` where appropriate.
* Remove starting `_` in some names (because they are reserved by the compiler).
* Clean up commented out code and unused definitions.

### CODE REVIEW

File: src/xrCore/Animation/Bone.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.cpp:45]:
```suggestion
        // направление вращения в ОДЕ отличается от направления вращения в X-Ray,
        // поэтому меняем знак у лимитов
```

Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.cpp]: Revert formatting.
```suggestion
        // Kostya Slipchenko say:
        // направление вращения в ОДЕ отличается от направления вращения в X-Ray,
        // поэтому меняем знак у лимитов
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp:505]: Make enum to be u32 then!
```
enum EBoneCallbackType : u32
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]: Return

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]:
```suggestion
    [[nodiscard]] IC BOOL callback_overwrite() const { return Callback_overwrite; } // performance hint - don't calc anims
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]: Return

File: src/xrCore/Animation/BoneEditor.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/BoneEditor.cpp]: Return deleted code **with** the original formatting.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/BoneEditor.cpp]: Default not needed

File: src/xrCore/Animation/BoneEditor.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/BoneEditor.cpp]: Return deleted code **with** the original formatting.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/BoneEditor.cpp]: There are several reformattings like `case X::Y: {` done. Brackets should always be on next line:
```suggestion
    case SBoneShape::stBox:
    {
```

File: src/xrCore/Animation/BoneEditor.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/BoneEditor.cpp]:
```suggestion
    case SBoneShape::stCylinder: shape.cylinder.m_center.add(amount); break;
```

File: src/xrCore/Animation/BoneEditor.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/BoneEditor.cpp]: Return deleted code **with** the original formatting.

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]:
```suggestion
    virtual size_t mem_usage()
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]:
```suggestion
        size_t sz = sizeof(*this) + sizeof(vecBones::value_type) * children.size();
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]:
```suggestion
    [[nodiscard]] u16 GetNumChildren() const override { return (u16)children.size(); } // XXX: maybe change return type to size_t instead of u16. Check later.
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]: Should we add [[nodiscard]] here?

File: src/xrCore/Animation/Bone.cpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.cpp]: Reverted.

File: src/xrCore/Animation/Bone.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.cpp]:
```suggestion
static const Fobb dummy = Fobb().identity(); // NOLINT
```

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp:505]: Underlying type is fixed to `u32` now.

File: src/xrCore/Animation/Bone.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.cpp]: Explicit type

File: src/xrCore/Animation/Bone.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.cpp]: Explicit type

File: src/xrCore/Animation/Bone.cpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.cpp]: Just a typo
```suggestion
        // F.w_float (std::min(-joint_limit.limit.x, -joint_limit.limit.y)); // min (swap special for ODE)
```

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: This is clang-format's work. If you prefer this formatting over current, we need to adjust `.clang-format` config.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (TmLev) [reply src/xrCore/Animation/BoneEditor.cpp]: Also `clang-format`'s fault.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: image

File: src/xrCore/Animation/Bone.cpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.cpp]: Changed.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Well, probably. I wasn't sure if it was necessary.

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [reply src/xrCore/Animation/Bone.hpp]: But you can remove additional blank line between them, like this:
```
    // Fmatrix last_transform;
    // Fmatrix render_transform;
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp:364]: Return removed comment, but you can format it:
```
// { return last_transform; }
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp:366]: Return removed comment, but you can format it:
```
// { return render_transform; }
```

File: src/xrCore/Animation/Bone.cpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.cpp]: Done.

File: src/xrCore/Animation/Bone.cpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.cpp]: Done.

File: src/xrCore/Animation/Bone.cpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.cpp]: Fixed.

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp:462]: I'm suspicious about this. Some code may be dependent on empty constructor. (because empty constructor doesn't default initialize variables)

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [reply src/xrCore/Animation/Bone.hpp]: No, size_t is considered as built-in C++ type. Probably, we can workaround this in another way.

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]: Explicit type

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]: We don't even need that now 😉 
```suggestion
    bctDummy,
```

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp:459]: The default constructor doesn't implicitly initialize fields also. The only difference is that `CBoneData` may be trivial to construct now.

Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp:462]: This is a destructor, not a constructor. I'm not sure what are you trying to say.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: I think it is not. [Proof](https://godbolt.org/z/r7r6v1). Some `#include` should be able to fix this.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Done.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Right, removed.

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [reply src/xrCore/Animation/Bone.hpp:462]: Ahh, nevermind then.

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [reply src/xrCore/Animation/Bone.hpp]: Well, agree that it is `std` type, not built-in... But we use `size_t` instead of `std::size_t`, so any issues with that should be fixed in another way and in another pull request.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp:364]: Restored.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp:366]: Restored.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Added.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (TmLev) [reply src/xrCore/Animation/BoneEditor.cpp]: Restored.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (TmLev) [reply src/xrCore/Animation/BoneEditor.cpp]: Restored.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Restored.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Formatted.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Restored.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (TmLev) [reply src/xrCore/Animation/BoneEditor.cpp]: Restored.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (TmLev) [reply src/xrCore/Animation/BoneEditor.cpp]: Reformatted.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (TmLev) [reply src/xrCore/Animation/BoneEditor.cpp]: Changed.

File: src/xrCore/Animation/BoneEditor.cpp
Comment (TmLev) [reply src/xrCore/Animation/BoneEditor.cpp]: Restored.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Restored.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Restored.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Added.

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]: It is probably in the SDK, so, a little correction :)
```suggestion
    bool ExportOGF(IWriter& F); // XXX: implementation missing. Probably it is in the SDK code?
```

File: src/xrCore/Animation/Bone.hpp
Comment (Xottab-DUTY) [inline src/xrCore/Animation/Bone.hpp]: Why you have restored this field?
```suggestion
```

**Review by Xottab-DUTY — APPROVED** [review]

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Because I'm under the impression that you are not allowed to delete unused code.

Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Should I remove this?

Comment (Xottab-DUTY) [reply src/xrCore/Animation/Bone.hpp]: Sorry. 😄 It's not that hard that you just can't touch anything, you can!... But I'm just very accurate with the old code.

Comment (Xottab-DUTY) [reply src/xrCore/Animation/Bone.hpp]: > Should I remove this?

Yes. It wasn't intended for use anyway and was only needed for making this enum 4 bytes in size. Since we already made it u32 using underlying type, it is not needed.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Clarified.

File: src/xrCore/Animation/Bone.hpp
Comment (TmLev) [reply src/xrCore/Animation/Bone.hpp]: Removed.

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by Xottab-DUTY** [comment]
Thank you for the effort and patience!

Will merge right after CI build finishes.

**Pull request merged**
